### PR TITLE
=htc make HTTP compile under Scala 2.12.0-M3

### PR DESCRIPTION
**Pull request by ktoso**
Makes Akka HTTP compile under `2.12.0-M3`

### CODE REVIEW

**Comment by drewhk** [comment]
LGTM

**Comment by ktoso** [comment]
Though it'll break 2.11 (the macro...) in this state.

**Comment by ktoso** [comment]
Updated PR to survive on both 2.11 and 2.12

**Comment by 2m** [comment]
LGTM. Do the akka-parsing changes come from the parboiled2 upstream?

**Comment by ktoso** [comment]
Nope, it's a fix I found on my own / with @jrudolph. It possibly should be also in parboiled2

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/CacheDirective.scala [hunk 1]
```diff
@@ -96,12 +96,12 @@ object CacheDirectives {
 
   // http://tools.ietf.org/html/rfc7234#section-5.2.2.6
   final case class `private`(fieldNames: immutable.Seq[String]) extends FieldNamesDirective with ResponseDirective
-  object `private` {
-    @varargs def apply(fieldNames: String*): `private` = apply(immutable.Seq(fieldNames: _*))
+  case object `private` {
```
Comment (jrudolph) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/CacheDirective.scala]: `case object` is not strictly needed for `private`. In the `no-cache` case the companion object can actually be used as a standalone value.

Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/CacheDirective.scala]: Thanks, good catch - removed `case`

**Comment by ktoso** [comment]
Addressed comments and rebased, will merge after validation

**Comment by drewhk** [comment]
LGTM

**Comment by 2m** [comment]
Fixes #19809.

**Comment by ktoso** [comment]
It's time to merge this fellow ;)

**Pull request merged**